sys/xtimer/xtimer.c

```diff
@@ -233,8 +233,12 @@ static void _mutex_timeout(void *arg)
     mutex_thread_t *mt = (mutex_thread_t *)arg;
 
     mt->timeout = 1;
+    list_node_t *node = list_remove(&mt->mutex->queue,
+                                    (list_node_t *)&mt->thread->rq_entry);
+    if ((node != NULL) && (mt->mutex->queue.next == NULL)) {
+        mt->mutex->queue.next = MUTEX_LOCKED;
```

Comment:
Exactly.

When the mutex is unlocked, `queue.next` is NULL. ` mutex -> NULL`
When the mutex is locked by 1 thread `queue.next` is `MUTEX_LOCKED`.  `mutex -> LOCKED`
When the mutex is locked by 2+ threads `queue.next` is the `mt->thread->rq_entry` of a thread and the last `.next` is NULL. `mutex -> thread -> (thread -> ...) NULL`

If `node != NULL` we were removed from the list, hence 2+ threads where in the list. At the same time if `queue.next` is NULL means that we were the last thread in the list and we need to set `queue.next` to `MUTEX_LOCKED` as there's now 1 thread still owning the mutex.